mvcc/kvstore.go

```diff
@@ -160,6 +166,42 @@ func (s *store) Hash() (hash uint32, revision int64, err error) {
 	return h, s.currentRev, err
 }
 
+func (s *store) HashByRev(rev int64) (hash uint32, compactRev int64, err error) {
+	for {
+		s.mu.RLock()
+		compactMainRev := s.compactMainRev
+		currentRev := s.currentRev
+		compactc := s.compactc
+		s.mu.RUnlock()
+		if rev < compactMainRev {
+			return 0, compactMainRev, ErrCompacted
+		} else if rev > currentRev {
+			return 0, compactMainRev, ErrFutureRev
+		}
+		select {
+		case <-compactc:
+		case <-s.stopc:
+			return 0, 0, ErrClosed
+		}
+		s.mu.Lock()
+		if s.compactc == compactc {
+			defer s.mu.Unlock()
```

Comment:
probably need to retrieve  `currentRev = s.currentRev` here because it is under the `s.mu.Lock()` which marks the beginning KV state of `HashBucket` operation.